{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Links\n",
    "#LINK1: https://www.ncbi.nlm.nih.gov/home/develop/api/\n",
    "#LINK2: https://colab.research.google.com/drive/1VOuvANFR08twLBROqYwO_TV34pazgF0_\n",
    "#LINK3: https://www.nlm.nih.gov/pubs/techbull/mj12/mj12_pm_author_ranking.html\n",
    "#LINK4: https://pubmed.ncbi.nlm.nih.gov/?term=Kumar+V&cauthor_id=32489811\n",
    "#LINK5: https://www.ncbi.nlm.nih.gov/pmc/pmctopmid/#converter\n",
    "#\n",
    "#Notes:\n",
    "# Relevant:\n",
    "#   The current preconditons include having the Name and Organization previously, with the additional constraint\n",
    "# of a required computed author id that is generated from PubMed.\n",
    "#   The date has the months in 3/4 letter abbreviations for some ungodly reason. Will have to convert later.\n",
    "#   As of now, I return the DOI link, rather than the study it actually leads to. Ran out of time, but I think\n",
    "# this is easily fixable.\n",
    "#   The topics generated are the keywords from each article pulled. This is not a good way to do this, but should\n",
    "# be fine for a start. There exists MeSH headings on some studies that will work much better for directly pulling,\n",
    "# but ideally we just do something else entirely (Some early NLP? Maybe we just don't use tags here)\n",
    "#\n",
    "# Other:\n",
    "#   There exists an 'affliation' tag on each study published that lists the author's university/location\n",
    "# and potentially an email, which could be useful.\n",
    "#use doi to check if a study exists on pubmed\n",
    "#add pmid and check for it\n",
    "\n",
    "from bs4 import BeautifulSoup\n",
    "import requests\n",
    "import urllib.request\n",
    "import json\n",
    "import time"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'https://stackoverflow.com/'"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Converts any possible PubMed spelling of the month to a numerical we want\n",
    "# If the month is already a number, leave it alone\n",
    "# Add more entries if encountered\n",
    "def monthToNum(month):\n",
    "    if month == 'Jan' or month == 'January':\n",
    "        month = '01'\n",
    "    elif month == 'Feb' or month == 'February':\n",
    "        month = '02'\n",
    "    elif month == 'Mar' or month == 'March':\n",
    "        month = '03'\n",
    "    elif month == 'Apr' or month == 'April':\n",
    "        month = '04'\n",
    "    elif month == 'May':\n",
    "        month = '05'\n",
    "    elif month == 'Jun' or month == 'June':\n",
    "        month = '06'\n",
    "    elif month == 'Jul' or month == 'July':\n",
    "        month = '07'\n",
    "    elif month == 'Aug' or month == 'August':\n",
    "        month = '08'\n",
    "    elif month == 'Sep' or month == 'Sept' or month == 'September':\n",
    "        month = '09'\n",
    "    elif month == 'Oct' or month == 'October':\n",
    "        month = '10'\n",
    "    elif month == 'Nov' or month == 'November':\n",
    "        month = '11'\n",
    "    elif month == 'Dec' or month == 'December':\n",
    "        month = '12'\n",
    "    return month"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def generate_article_and_tags(soup):\n",
    "    \n",
    "    #######################\n",
    "    ### Article Section ###\n",
    "    #######################\n",
    "    \n",
    "    # Create a base article dictionary and populate it with the info found below\n",
    "    # If the majority of the fields are not replaced, something is probably going wrong\n",
    "    article = {'title': 'None', 'publication date': 'None', 'pdf link': 'None', 'description': 'None'}\n",
    "    \n",
    "    ### Article Title ###\n",
    "    # Search for the title of the article\n",
    "    title = soup.find('articletitle')\n",
    "    if title:\n",
    "        article['title'] = title.text\n",
    "    \n",
    "    ### Publication Date ###\n",
    "    # Search for the date of publication\n",
    "    # Sometimes only part of the date has been provided\n",
    "    pubdate = soup.find('pubdate')\n",
    "    date = 'None'\n",
    "    if pubdate:\n",
    "        date = pubdate.year.text\n",
    "        if pubdate.month:\n",
    "            month = monthToNum(pubdate.month.text)       \n",
    "            date += '/' + pubdate.month.text\n",
    "            if pubdate.day:\n",
    "                date += '/' + pubdate.day.text\n",
    "    article['publication date'] = date\n",
    "    \n",
    "    ### Pdf Link ###\n",
    "    #Search for DOI name (All published articles should have a DOI name)\n",
    "    doi = soup.find('articleid', idtype = \"doi\")\n",
    "    if doi:\n",
    "        #The pdf link should be the final redirect of accessing DOI name\n",
    "        pdflink = urllib.request.urlopen('https://doi.org/' + doi.text)\n",
    "        article['pdf link'] = pdflink.geturl()\n",
    "    \n",
    "    ### Abstract ###\n",
    "    #Search for article abstract (PubMed does not have any other form of description)\n",
    "    abst = soup.find('abstracttext')\n",
    "    if desc:\n",
    "        article['description'] = abst.text\n",
    "    \n",
    "    ####################\n",
    "    ### Tags Section ###\n",
    "    ####################\n",
    "    \n",
    "    #Temporary implementation of tag collection\n",
    "    tags = []\n",
    "    keywords = soup.find_all('keyword')\n",
    "    for keyword in keywords:\n",
    "        tags.append(keyword.text) \n",
    "    #Could add MeSH tags if exist, would probably be better than these\n",
    "    \n",
    "    return study, tags"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def checkPmid(pmid):\n",
    "    url = \"http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=pubmed&retmode=xml&id=['\" + pmid + \"']\"\n",
    "    #wait actually we dont have to do this, but i guess double checking would be helpful\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def generate_researcher(name, org='None', email='None', studyNum = 5, doi = 'None', pmid = 'None'):\n",
    "    \n",
    "    #######################\n",
    "    ### Link Generation ###\n",
    "    #######################\n",
    "    \n",
    "    ### PMID takes priority over DOI ###\n",
    "    #check doi\n",
    "    doi_to_pmi = \"https://www.ncbi.nlm.nih.gov/pmc/utils/idconv/v1.0/\"\n",
    "    \n",
    "    ### Search Link Creation ###\n",
    "    base = \"https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esearch.fcgi?db=pubmed&retmode=json\"\n",
    "    #Search \n",
    "    #Name = Name of researcher\n",
    "    #StudyNum = Max number of result\n",
    "    \n",
    "    #Search Query - reorganize name into the format 'Last+F', though I think including the entire first name also works\n",
    "    query = name.split()[1] + \"+\" + name.split()[0]\n",
    "    #Computed author id\n",
    "    #cauthor = '32489811'\n",
    "    #Assemble elink\n",
    "    url = base + \"&retmax=\" + studyNum + \"&term=\" + query# + \"&cauthor_id=\" + cauthor\n",
    "    #Test esearch URL\n",
    "    #print(url)\n",
    "    \n",
    "    ### Create List of Studies ###\n",
    "    #Add search results to a list\n",
    "    webpage = urllib.request.urlopen(url).read()\n",
    "    dict_page =json.loads(webpage)\n",
    "    idlist = dict_page[\"esearchresult\"][\"idlist\"]\n",
    "    #Test for search results\n",
    "    print(idlist)\n",
    "    \n",
    "    #############################\n",
    "    ### Researcher Generation ###\n",
    "    #############################\n",
    "    \n",
    "    #Setup dictionary\n",
    "    researcher = {'name': name, 'organization': org, 'topics': 'None', 'studies': 'None'}\n",
    "    studies = []\n",
    "    topics = []\n",
    "    \n",
    "    for link in idlist:\n",
    "        url = \"http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=pubmed&retmode=xml&id=idlist\"\n",
    "        url = url.replace('idlist', link)\n",
    "        \n",
    "        r = requests.get(url)\n",
    "        soup = BeautifulSoup(r.content, \"html.parser\")\n",
    "        print(soup)\n",
    "        \n",
    "        #study, tags = generate_study_and_tags(soup)\n",
    "        \n",
    "        #Occationally I ran into blank studies? I think this was a result of other issues with my code\n",
    "        ADD = False\n",
    "        for entries in study.values():\n",
    "            if entries != 'None':\n",
    "                ADD = True\n",
    "        if ADD:\n",
    "            studies.append(study)\n",
    "            topics = topics + tags\n",
    "        \n",
    "        #Need to sleep otherwise we get request error\n",
    "        time.sleep(0.1)\n",
    "    \n",
    "    if studies:\n",
    "        #Sort the studies by date\n",
    "        studies = sorted(studies, key = lambda i: i['publication date'], reverse=True)\n",
    "        researcher['studies'] = studies\n",
    "    if topics:\n",
    "        researcher['topics'] = topics\n",
    "    #Test scraping results\n",
    "    #print(researcher)\n",
    "    \n",
    "    return researcher"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['33653945', '33057063', '32529416', '32378039', '31933147']\n",
      "<?xml version=\"1.0\" ?>\n",
      "<!DOCTYPE PubmedArticleSet PUBLIC \"-//NLM//DTD PubMedArticle, 1st January 2019//EN\" \"https://dtd.nlm.nih.gov/ncbi/pubmed/out/pubmed_190101.dtd\">\n",
      "\n",
      "<pubmedarticleset>\n",
      "<pubmedarticle>\n",
      "<medlinecitation owner=\"NLM\" status=\"In-Data-Review\">\n",
      "<pmid version=\"1\">33653945</pmid>\n",
      "<daterevised>\n",
      "<year>2021</year>\n",
      "<month>03</month>\n",
      "<day>03</day>\n",
      "</daterevised>\n",
      "<article pubmodel=\"Print\">\n",
      "<journal>\n",
      "<issn issntype=\"Electronic\">1538-8514</issn>\n",
      "<journalissue citedmedium=\"Internet\">\n",
      "<volume>20</volume>\n",
      "<issue>3</issue>\n",
      "<pubdate>\n",
      "<year>2021</year>\n",
      "<month>Mar</month>\n",
      "</pubdate>\n",
      "</journalissue>\n",
      "<title>Molecular cancer therapeutics</title>\n",
      "<isoabbreviation>Mol Cancer Ther</isoabbreviation>\n",
      "</journal>\n",
      "<articletitle>Resistance to Pyrrolobenzodiazepine Dimers Is Associated with SLFN11 Downregulation and Can Be Reversed through Inhibition of ATR.</articletitle>\n",
      "<pagination>\n",
      "<medlinepgn>541-552</medlinepgn>\n",
      "</pagination>\n",
      "<elocationid eidtype=\"doi\" validyn=\"Y\">10.1158/1535-7163.MCT-20-0351</elocationid>\n",
      "<abstract>\n",
      "<abstracttext>Resistance to antibody-drug conjugates (ADCs) has been observed in both preclinical models and clinical studies. However, mechanisms of resistance to pyrrolobenzodiazepine (PBD)-conjugated ADCs have not been well characterized and thus, this study was designed to investigate development of resistance to PBD dimer warheads and PBD-conjugated ADCs. We established a PBD-resistant cell line, 361-PBDr, by treating human breast cancer MDA-MB-361 cells with gradually increasing concentrations of SG3199, the PBD dimer released from the PBD drug-linker tesirine. 361-PBDr cells were over 20-fold less sensitive to SG3199 compared with parental cells and were cross-resistant to other PBD warhead and ADCs conjugated with PBDs. Proteomic profiling revealed that downregulation of Schlafen family member 11 (SLFN11), a putative DNA/RNA helicase, sensitizing cancer cells to DNA-damaging agents, was associated with PBD resistance. Confirmatory studies demonstrated that siRNA knockdown of SLFN11 in multiple tumor cell lines conferred reduced sensitivity to SG3199 and PBD-conjugated ADCs. Treatment with EPZ011989, an EZH2 inhibitor, derepressed SLFN11 expression in 361-PBDr and other SLFN11-deficient tumor cells, and increased sensitivity to PBD and PBD-conjugated ADCs, indicating that the suppression of SLFN11 expression is associated with histone methylation as reported. Moreover, we demonstrated that combining an ataxia telangiectasia and Rad3-related protein (ATR) inhibitor, AZD6738, with SG3199 or PBD-based ADCs led to synergistic cytotoxicity in either resistant 361-PBDr cells or cells that SLFN11 was knocked down via siRNA. Collectively, these data provide insights into potential development of resistance to PBDs and PBD-conjugated ADCs, and more importantly, inform strategy development to overcome such resistance.</abstracttext>\n",
      "<copyrightinformation>©2021 American Association for Cancer Research.</copyrightinformation>\n",
      "</abstract>\n",
      "<authorlist completeyn=\"Y\">\n",
      "<author validyn=\"Y\">\n",
      "<lastname>Mao</lastname>\n",
      "<forename>Shenlan</forename>\n",
      "<initials>S</initials>\n",
      "<affiliationinfo>\n",
      "<affiliation>Early Oncology R&amp;D, AstraZeneca, Gaithersburg, Maryland. shenlan.mao@astrazeneca.com john.harper@astrazeneca.com.</affiliation>\n",
      "</affiliationinfo>\n",
      "</author>\n",
      "<author validyn=\"Y\">\n",
      "<lastname>Chaerkady</lastname>\n",
      "<forename>Raghothama</forename>\n",
      "<initials>R</initials>\n",
      "<affiliationinfo>\n",
      "<affiliation>Antibody Discovery &amp; Protein Engineering, AstraZeneca, Gaithersburg, Maryland.</affiliation>\n",
      "</affiliationinfo>\n",
      "</author>\n",
      "<author validyn=\"Y\">\n",
      "<lastname>Yu</lastname>\n",
      "<forename>Wen</forename>\n",
      "<initials>W</initials>\n",
      "<affiliationinfo>\n",
      "<affiliation>Bioinformatics, AstraZeneca, Gaithersburg, Maryland.</affiliation>\n",
      "</affiliationinfo>\n",
      "</author>\n",
      "<author validyn=\"Y\">\n",
      "<lastname>D'Angelo</lastname>\n",
      "<forename>Gina</forename>\n",
      "<initials>G</initials>\n",
      "<affiliationinfo>\n",
      "<affiliation>Biometrics, AstraZeneca, Gaithersburg, Maryland.</affiliation>\n",
      "</affiliationinfo>\n",
      "</author>\n",
      "<author validyn=\"Y\">\n",
      "<lastname>Garcia</lastname>\n",
      "<forename>Andrew</forename>\n",
      "<initials>A</initials>\n",
      "<affiliationinfo>\n",
      "<affiliation>Antibody Discovery &amp; Protein Engineering, AstraZeneca, Gaithersburg, Maryland.</affiliation>\n",
      "</affiliationinfo>\n",
      "</author>\n",
      "<author validyn=\"Y\">\n",
      "<lastname>Chen</lastname>\n",
      "<forename>Hong</forename>\n",
      "<initials>H</initials>\n",
      "<affiliationinfo>\n",
      "<affiliation>Early Oncology R&amp;D, AstraZeneca, Gaithersburg, Maryland.</affiliation>\n",
      "</affiliationinfo>\n",
      "</author>\n",
      "<author validyn=\"Y\">\n",
      "<lastname>Barrett</lastname>\n",
      "<forename>Allison M</forename>\n",
      "<initials>AM</initials>\n",
      "<affiliationinfo>\n",
      "<affiliation>Early Oncology R&amp;D, AstraZeneca, Gaithersburg, Maryland.</affiliation>\n",
      "</affiliationinfo>\n",
      "</author>\n",
      "<author validyn=\"Y\">\n",
      "<lastname>Phipps</lastname>\n",
      "<forename>Sandrina</forename>\n",
      "<initials>S</initials>\n",
      "<affiliationinfo>\n",
      "<affiliation>Antibody Discovery &amp; Protein Engineering, AstraZeneca, Gaithersburg, Maryland.</affiliation>\n",
      "</affiliationinfo>\n",
      "</author>\n",
      "<author validyn=\"Y\">\n",
      "<lastname>Fleming</lastname>\n",
      "<forename>Ryan</forename>\n",
      "<initials>R</initials>\n",
      "<affiliationinfo>\n",
      "<affiliation>Antibody Discovery &amp; Protein Engineering, AstraZeneca, Gaithersburg, Maryland.</affiliation>\n",
      "</affiliationinfo>\n",
      "</author>\n",
      "<author validyn=\"Y\">\n",
      "<lastname>Hess</lastname>\n",
      "<forename>Sonja</forename>\n",
      "<initials>S</initials>\n",
      "<identifier source=\"ORCID\">https://orcid.org/0000-0002-5904-9816</identifier>\n",
      "<affiliationinfo>\n",
      "<affiliation>Antibody Discovery &amp; Protein Engineering, AstraZeneca, Gaithersburg, Maryland.</affiliation>\n",
      "</affiliationinfo>\n",
      "</author>\n",
      "<author validyn=\"Y\">\n",
      "<lastname>Koopmann</lastname>\n",
      "<forename>Jens-Oliver</forename>\n",
      "<initials>JO</initials>\n",
      "<affiliationinfo>\n",
      "<affiliation>Early Oncology R&amp;D, AstraZeneca, Gaithersburg, Maryland.</affiliation>\n",
      "</affiliationinfo>\n",
      "</author>\n",
      "<author validyn=\"Y\">\n",
      "<lastname>Dimasi</lastname>\n",
      "<forename>Nazzareno</forename>\n",
      "<initials>N</initials>\n",
      "<affiliationinfo>\n",
      "<affiliation>Antibody Discovery &amp; Protein Engineering, AstraZeneca, Gaithersburg, Maryland.</affiliation>\n",
      "</affiliationinfo>\n",
      "</author>\n",
      "<author validyn=\"Y\">\n",
      "<lastname>Wilson</lastname>\n",
      "<forename>Susan</forename>\n",
      "<initials>S</initials>\n",
      "<affiliationinfo>\n",
      "<affiliation>Antibody Discovery &amp; Protein Engineering, AstraZeneca, Gaithersburg, Maryland.</affiliation>\n",
      "</affiliationinfo>\n",
      "</author>\n",
      "<author validyn=\"Y\">\n",
      "<lastname>Pugh</lastname>\n",
      "<forename>Kathryn</forename>\n",
      "<initials>K</initials>\n",
      "<affiliationinfo>\n",
      "<affiliation>Spirogen, AstraZeneca, London, United Kingdom.</affiliation>\n",
      "</affiliationinfo>\n",
      "</author>\n",
      "<author validyn=\"Y\">\n",
      "<lastname>Cook</lastname>\n",
      "<forename>Kimberly</forename>\n",
      "<initials>K</initials>\n",
      "<affiliationinfo>\n",
      "<affiliation>Early Oncology R&amp;D, AstraZeneca, Gaithersburg, Maryland.</affiliation>\n",
      "</affiliationinfo>\n",
      "</author>\n",
      "<author validyn=\"Y\">\n",
      "<lastname>Masterson</lastname>\n",
      "<forename>Luke A</forename>\n",
      "<initials>LA</initials>\n",
      "<affiliationinfo>\n",
      "<affiliation>Spirogen, AstraZeneca, London, United Kingdom.</affiliation>\n",
      "</affiliationinfo>\n",
      "</author>\n",
      "<author validyn=\"Y\">\n",
      "<lastname>Gao</lastname>\n",
      "<forename>Changshou</forename>\n",
      "<initials>C</initials>\n",
      "<affiliationinfo>\n",
      "<affiliation>Antibody Discovery &amp; Protein Engineering, AstraZeneca, Gaithersburg, Maryland.</affiliation>\n",
      "</affiliationinfo>\n",
      "</author>\n",
      "<author validyn=\"Y\">\n",
      "<lastname>Wu</lastname>\n",
      "<forename>Herren</forename>\n",
      "<initials>H</initials>\n",
      "<affiliationinfo>\n",
      "<affiliation>Antibody Discovery &amp; Protein Engineering, AstraZeneca, Gaithersburg, Maryland.</affiliation>\n",
      "</affiliationinfo>\n",
      "</author>\n",
      "<author validyn=\"Y\">\n",
      "<lastname>Herbst</lastname>\n",
      "<forename>Ronald</forename>\n",
      "<initials>R</initials>\n",
      "<affiliationinfo>\n",
      "<affiliation>Early Oncology R&amp;D, AstraZeneca, Gaithersburg, Maryland.</affiliation>\n",
      "</affiliationinfo>\n",
      "</author>\n",
      "<author validyn=\"Y\">\n",
      "<lastname>Howard</lastname>\n",
      "<forename>Philip W</forename>\n",
      "<initials>PW</initials>\n",
      "<affiliationinfo>\n",
      "<affiliation>Spirogen, AstraZeneca, London, United Kingdom.</affiliation>\n",
      "</affiliationinfo>\n",
      "</author>\n",
      "<author validyn=\"Y\">\n",
      "<lastname>Tice</lastname>\n",
      "<forename>David A</forename>\n",
      "<initials>DA</initials>\n",
      "<affiliationinfo>\n",
      "<affiliation>Early Oncology R&amp;D, AstraZeneca, Gaithersburg, Maryland.</affiliation>\n",
      "</affiliationinfo>\n",
      "</author>\n",
      "<author validyn=\"Y\">\n",
      "<lastname>Cobbold</lastname>\n",
      "<forename>Mark</forename>\n",
      "<initials>M</initials>\n",
      "<affiliationinfo>\n",
      "<affiliation>Early Oncology R&amp;D, AstraZeneca, Gaithersburg, Maryland.</affiliation>\n",
      "</affiliationinfo>\n",
      "</author>\n",
      "<author validyn=\"Y\">\n",
      "<lastname>Harper</lastname>\n",
      "<forename>Jay</forename>\n",
      "<initials>J</initials>\n",
      "<affiliationinfo>\n",
      "<affiliation>Early Oncology R&amp;D, AstraZeneca, Gaithersburg, Maryland. shenlan.mao@astrazeneca.com john.harper@astrazeneca.com.</affiliation>\n",
      "</affiliationinfo>\n",
      "</author>\n",
      "</authorlist>\n",
      "<language>eng</language>\n",
      "<publicationtypelist>\n",
      "<publicationtype ui=\"D016428\">Journal Article</publicationtype>\n",
      "</publicationtypelist>\n",
      "</article>\n",
      "<medlinejournalinfo>\n",
      "<country>United States</country>\n",
      "<medlineta>Mol Cancer Ther</medlineta>\n",
      "<nlmuniqueid>101132535</nlmuniqueid>\n",
      "<issnlinking>1535-7163</issnlinking>\n",
      "</medlinejournalinfo>\n",
      "<citationsubset>IM</citationsubset>\n",
      "</medlinecitation>\n",
      "<pubmeddata>\n",
      "<history>\n",
      "<pubmedpubdate pubstatus=\"received\">\n",
      "<year>2020</year>\n",
      "<month>06</month>\n",
      "<day>10</day>\n",
      "</pubmedpubdate>\n",
      "<pubmedpubdate pubstatus=\"revised\">\n",
      "<year>2020</year>\n",
      "<month>11</month>\n",
      "<day>01</day>\n",
      "</pubmedpubdate>\n",
      "<pubmedpubdate pubstatus=\"accepted\">\n",
      "<year>2021</year>\n",
      "<month>01</month>\n",
      "<day>07</day>\n",
      "</pubmedpubdate>\n",
      "<pubmedpubdate pubstatus=\"entrez\">\n",
      "<year>2021</year>\n",
      "<month>3</month>\n",
      "<day>3</day>\n",
      "<hour>5</hour>\n",
      "<minute>39</minute>\n",
      "</pubmedpubdate>\n",
      "<pubmedpubdate pubstatus=\"pubmed\">\n",
      "<year>2021</year>\n",
      "<month>3</month>\n",
      "<day>4</day>\n",
      "<hour>6</hour>\n",
      "<minute>0</minute>\n",
      "</pubmedpubdate>\n",
      "<pubmedpubdate pubstatus=\"medline\">\n",
      "<year>2021</year>\n",
      "<month>3</month>\n",
      "<day>4</day>\n",
      "<hour>6</hour>\n",
      "<minute>0</minute>\n",
      "</pubmedpubdate>\n",
      "</history>\n",
      "<publicationstatus>ppublish</publicationstatus>\n",
      "<articleidlist>\n",
      "<articleid idtype=\"pubmed\">33653945</articleid>\n",
      "<articleid idtype=\"pii\">20/3/541</articleid>\n",
      "<articleid idtype=\"doi\">10.1158/1535-7163.MCT-20-0351</articleid>\n",
      "</articleidlist>\n",
      "<referencelist>\n",
      "<reference>\n",
      "<citation>Khongorzul P, Ling CJ, Khan FU, Ihsan AU, Zhang J. Antibody-drug conjugates: a comprehensive review. Mol Cancer Res. 2020;18:3–19.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Al-Rohil RN, Torres-Cabala CA, Patel A, Tetzlaff MT, Ivan D, Nagarajan P, et al. Loss of CD30 expression after treatment with brentuximab vedotin in a patient with anaplastic large cell lymphoma: a novel finding. J Cutan Pathol. 2016;43:1161–66.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Linenberger ML, Hong T, Flowers D, Sievers EL, Gooley TA, Bennett JM, et al. Multidrug-resistance phenotype and clinical responses to gemtuzumab ozogamicin. Blood. 2001;98:988–94.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Takeshita A, Shinjo K, Yamakage N, Ono T, Hirano I, Matsui H, et al. CMC-544 (inotuzumab ozogamicin) shows less effect on multidrug resistant cells: analyses in cell lines and cells from patients with B-cell chronic lymphocytic leukaemia and lymphoma. Br J Haematol. 2009;146:34–43.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Li G, Guo J, Shen BQ, Yadav DB, Sliwkowski MX, Crocker LM, et al. Mechanisms of acquired resistance to trastuzumab emtansine in breast cancer cells. Mol Cancer Ther. 2018;17:1441–53.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Chen R, Hou J, Newman E, Kim Y, Donohue C, Liu X, et al. CD30 downregulation, MMAE resistance, and MDR1 upregulation are all associated with resistance to brentuximab vedotin. Mol Cancer Ther. 2015;14:1376–84.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Tiberghien AC, Levy JN, Masterson LA, Patel NV, Adams LR, Corbett S, et al. Design and synthesis of tesirine, a clinical antibody−drug conjugate pyrrolobenzodiazepine dimer payload. ACS Med Chem Lett. 2016;24:983–87.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Hartley JA, Flynn MJ, Bingham JP, Corbett S, Reinert H, Tiberghien A, et al. Pre-clinical pharmacology and mechanism of action of SG3199, the pyrrolobenzodiazepine (PBD) dimer warhead component of antibody-drug conjugate (ADC) payload tesirine. Sci Rep. 2018;8:10479.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Mantaj J, Jackson PJM, Rahman KM, Thurston DE. From anthramycin to pyrrolobenzodiazepine (PBD)-containing antibody–drug conjugates (ADCs). Angew Chem Int Ed. 2017;56:462–88.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Lewis T, Olson DJ, Gordon KA, Sandall SL, Miyamoto J, Westendorf L, et al. SGN-CD352A: a novel humanized anti-CD352 antibody-drug conjugate for the treatment of multiple myeloma [abstract]. Proceedings of the 107th Annual Meeting of the American Association for Cancer Research; 2016 Apr 16–20; New Orleans, LA.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Pore N, Borrok MJ, Rebelatto M, Hinrichs MJ, Schifferli KP, Monks NR, et al. Discovery and development of MEDI7247, a novel pyrrolobenzodiazepine (PBD)-based antibody drug conjugate targeting ASCT2, for treating hematological and solid cancers [abstract]. Proceedings of the American Association for Cancer Research Annual Meeting 2018; 2018 Apr 14–18; Chicago, IL.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Kinneer K, Flynn M, Thomas SB, Meekin J, Varkey R, Xiao X, et al. Preclinical assessment of an antibody-PBD conjugate that targets BCMA on multiple myeloma and myeloma progenitor cells. Leukemia. 2019;33:766–71.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Tallman MS, Feingold JM, Spira AI, Rizzieri D, Atallah EL, Stock W, et al. A phase 1, open-label, dose-escalation, multicenter study to evaluate the tolerability, safety, pharmacokinetics, and activity of ADCT-301 in patients with relapsed or refractory CD25-positive acute myeloid leukemia. J Clin Oncol. 2016;34.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Zammarchi F, Corbett S, Adams L, Tyrer PC, Kiakos K, Janghra N, et al. ADCT-402, a PBD dimer-containing antibody drug conjugate targeting CD19-expressing malignancies. Blood. 2018;131:1094–105.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Corbett S, Huang S, Zammarchi F, Howard PW, Berkel PHV, Hartley JA. The role of specific ATP-binding cassette transporters in the acquired resistance to pyrrolobenzodiazepine dimer containing antibody-drug conjugates. Mol Cancer Ther. 2020;19:1856–65.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Kinneer K, Meekin J, Tiberghien AC, Tai YT, Phipps S, Kiefer CM, et al. SLC46A3 as a potential predictive biomarker for antibody-drug conjugates bearing noncleavable linked maytansinoid and pyrrolobenzodiazepine warheads. Clin Cancer Res. 2018;24:6570–82.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Zoppolia G, Regairaza M, Leoa E, Reinholda WC, Varma S, Ballestrerob A, et al. Putative DNA/RNA helicase Schlafen-11 (SLFN11) sensitizes cancer cells to DNA-damaging agents. Proc Natl Acad Sci U S A. 2012;109:15030–35.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Murai J, Thomas A, Miettinen M, Pommier Y. Schlafen 11 (SLFN11), a restriction factor for replicative stress induced by DNA-targeting anti-cancer therapies. Pharmacol Ther. 2019;201:94–102.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Murai J, Tang SW, Leo E, Baechler SA, Redon CE, Zhang H, et al. SLFN11 blocks stressed replication forks independently of ATR. Mol Cell. 2018;69:371–84.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Gardner EE, Lok BH, Schneeberger VE, Desmeules P, Miles LA, Arnold PK, et al. Chemosensitive relapse in small cell lung cancer proceeds through an EZH2-SLFN11 axis. Cancer Cell. 2017;31:286–99.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Foote KM, Nissink JWM, McGuire T, Turner P, Guichard S, Yates JWT, et al. Discovery and characterization of AZD6738, a potent inhibitor of ataxia telangiectasia mutated and Rad3 related (ATR) kinase with application as an anticancer agent. J Med Chem. 2018;61:9889–07.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Harper J, Lloyd C, Dimasi N, Toader D, Marwood R, Lewis L, et al. Preclinical evaluation of MEDI0641, a pyrrolobenzodiazepine-conjugated antibody-drug conjugate targeting 5T4. Mol Cancer Ther. 2017;16:1576–87.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Hartley JA, Spanswick VJ, Brooks N, Clingen PH, McHugh PJ, Hochhauser D, et al. SJG-136 (NSC 694501), a novel rationally designed DNA minor groove interstrand cross-linking agent with potent and broad spectrum antitumor activity: part 1: cellular pharmacology, in vitro and initial in vivo antitumor activity. Cancer Res. 2004;64:6693–99.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Li JY, Perry SR, Muniz-Medina V, Wang X, Wetzel LK, Rebelatto MC, et al. A biparatopic HER2-targeting antibody-drug conjugate induces tumor regression in primary models refractory to or ineligible for HER2-targeted therapy. Cancer Cell. 2016;29:117–29.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Ratovitski T, Chaerkady R, Kammers K, Stewart JC, Zavala A, Pletnikova‖ O, et al. Quantitative proteomic analysis reveals similarities between Huntington’s disease (HD) and Huntington’s disease-like 2 (HDL2) human brains. J Proteome Res. 2016;15:3266–83.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Brosch M, Yu L, Hubbard T, Choudhary J. Accurate and sensitive peptide identification with Mascot Percolator. J Proteome Res. 2009;8:3176–81.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>D’Angelo G, Chaerkady R, Yu W, Hizal DB, Hess S, Zhao W. Statistical models for the analysis of isobaric tags multiplexed quantitative proteomics. J Proteome Res. 2017;16:3124–36.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Tyanova S, Temu T, Sinitcyn P, Carlson A, Hein MY, Geiger T, et al. The Perseus computational platform for comprehensive analysis of (prote)omics data. Nat Methods. 2016;13:731–40.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Di Veroli GY, Fornari C, Wang D, Mollard S, Bramhall JL, Richards FM, et al. Combenefit: an interactive platform for the analysis and visualization of drug combinations. Bioinformatics. 2016;32:2866–68.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Liang DP, Huang TQ, Li SJ, Chen ZJ. Knockdown of S100A4 chemosensitizes human laryngeal carcinoma cells in vitro through inhibition of Slug. Eur Rev Med Pharmacol Sci. 2014;18:3484–90.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Nam KS, Son SH, Oh S, Jeon D, Kim H, Noh DY, et al. Binding of galectin-1 to integrin β1 potentiates drug resistance by promoting survivin expression in breast cancer cells. Oncotarget. 2017;8:35804–23.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Novotna R, Wsol V, Xiong G, Maser E. Inactivation of the anticancer drugs doxorubicin and oracin by aldo–keto reductase (AKR) 1C3. Toxicol Lett. 2008;181:1–6.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Takashima T, Sakamoto N, Murai J, Taniyama D, Honma R, Ukai S, et al. Immunohistochemical analysis of SLFN11 expression uncovers potential non-responders to DNA-damaging agents overlooked by tissue RNA-seq. Virchows Arch. 2020.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Tang SW, Thomas A, Murai J, Trepel JB, Bates SE, Rajapakse V, et al. Overcoming resistance to DNA-targeted agents by epigenetic activation of Schlafen 11 (SLFN11) expression with class I histone deacetylase inhibitors. Clin Cancer Res. 2018;24:1944–53.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Nogales V, Reinhold WC, Varma S, Martinez-Cardus A, Moutinho C, Moran S, et al. Epigenetic inactivation of the putative DNA/RNA helicase SLFN11 in human cancer confers resistance to platinum drugs. Oncotarget. 2015;7:3084–97.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Garner KM, Eastman A. Variations in Mre11/Rad50/Nbs1 status and DNA damage-induced S-phase arrest in the cell lines of the NCI60 panel. BMC Cancer. 2011;11:206.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Murai J, Feng Y, Yu GK, Ru Y, Tang SW, Shen Y, et al. Resistance to PARP inhibitors by SLFN11 inactivation can be overcome by ATR inhibition. Oncotarget. 2016;7:76534–50.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Coussy F, El-Botty R, Château-Joubert S, Dahmani A, Montaudon E, Leboucher S, et al. BRCAness, SLFN11, and RB1 loss predict response to topoisomerase I inhibitors in triple-negative breast cancers. Sci Transl Med. 2020;12:eaax2625.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Spanswick VJ, Craddock C, Sekhar M, Mahendra P, Shankaranarayana P, Hughes RG, et al. Repair of DNA interstrand crosslinks as a mechanism of clinical resistance to melphalan in multiple myeloma. Blood. 2002;100:224–29.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Deans AJ, West SC. DNA interstrand crosslink repair and cancer. Nat Rev Cancer. 2013;11:467–80.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Clingen PH, Silva IUD, McHugh PJ, Ghadessy FJ, Tilby MJ, Thurston DE, et al. The XPF-ERCC1 endonuclease and homologous recombination contribute to the repair of minor groove DNA interstrand crosslinks in mammalian cells produced by the pyrrolo[2,1-c][1,4]benzodiazepinedimer SJG-136. Nucleic Acids Res. 2005;33:3283–91.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Zhong H, Chen C, Tammali R, Breen S, Zhang J, Fazenbaker C, et al. Improved therapeutic window in BRCA-mutant tumors with antibody-linked pyrrolobenzodiazepine dimers with and without PARP inhibition. Mol Cancer Ther. 2019;18:89–99.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Hucl T, Rago C, Gallmeier E, Brody JR, Gorospe M, Kern SE. A syngeneic variance library for functional annotation of human variation: application to BRCA2. Cancer Res. 2008;68:5023–30.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Pietanza MC, Waqar SN, Krug LM, Dowlati A, Hann CL, Chiappori A, et al. Randomized, double-blind, phase II study of temozolomide in combination with either veliparib or placebo in patients with relapsed-sensitive or refractory small-cell lung cancer. J Clin Oncol. 2018;36:2386–94.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Zhao C, Takahashi N, Rajan A. Sleep better on combination therapy: SLFN11 predicts response to veliparib and temozolomide in recurrent small cell lung cancer. Transl Lung Cancer Res. 2018;7:S308–11.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Flynn RL, Zou L. ATR: a master conductor of cellular responses to DNA replication stress. Trends Biochem Sci. 2011;36:133–40.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Karnitz LM, Zou L. Molecular pathways: targeting ATR in cancer therapy. Clin Cancer Res. 2015;21:4780–85.</citation>\n",
      "</reference>\n",
      "<reference>\n",
      "<citation>Li M, Kao E, Malone D, Gao X, Wang JYJ, David M, et al. DNA damage-induced cell death relies on SLFN11-dependent cleavage of distinct type II tRNAs. Nat Struct Mol Biol. 2018;25:1047–58.</citation>\n",
      "</reference>\n",
      "</referencelist>\n",
      "</pubmeddata>\n",
      "</pubmedarticle>\n",
      "</pubmedarticleset>\n"
     ]
    },
    {
     "ename": "NameError",
     "evalue": "name 'study' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-8-5a6281916af9>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mgenerate_researcher\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'Andrew Garcia'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;32m<ipython-input-7-1b6fe63f200d>\u001b[0m in \u001b[0;36mgenerate_researcher\u001b[1;34m(name, org, email, studyNum, caid, pmid)\u001b[0m\n\u001b[0;32m     46\u001b[0m         \u001b[1;31m#Occationally I ran into blank studies? I think this was a result of other issues with my code\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     47\u001b[0m         \u001b[0mADD\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mFalse\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 48\u001b[1;33m         \u001b[1;32mfor\u001b[0m \u001b[0mentries\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mstudy\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mvalues\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     49\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mentries\u001b[0m \u001b[1;33m!=\u001b[0m \u001b[1;34m'None'\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     50\u001b[0m                 \u001b[0mADD\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mTrue\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mNameError\u001b[0m: name 'study' is not defined"
     ]
    }
   ],
   "source": [
    "print(generate_researcher('Andrew Garcia'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
